### PR TITLE
feat: HIP-904: Adapt creation flows to support unlimited auto associations and adjust fees

### LINKED ISSUE #13061: Adapt Creation flows to support the unlimited maxAutoAssociation and update fee calculations

### PR DESCRIPTION
**Pull request by stoyanov-st**
**Description**:
We are adding validations for `-1` maxAutoAssociations that acts as unlimited flag for auto associations.
We are also removing the fees for auto association slots if we have enabled the `unlimitedAutoAssociationsEnabled` flag.

**Related issue(s)**:

Fixes #13061 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/AutoAccountCreationSuite.java [hunk 1]
```diff
@@ -546,7 +549,8 @@ final HapiSpec multipleTokenTransfersSucceed() {
         final var initialTokenSupply = 1000;
         final var multiTokenXfer = "multiTokenXfer";
 
-        return defaultHapiSpec("multipleTokenTransfersSucceed")
+        return propertyPreservingHapiSpec("multipleTokenTransfersSucceed")
+                .preserving("entities.unlimitedAutoAssociations", FALSE)
```
Comment (kselveliev) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/AutoAccountCreationSuite.java:553]: I'm not sure if this is correct, I believe that in this case preserving only accepts list of names of configs. But seems like this change assumes that preserving accepts CONFIG_NAME -> CONFIG_VALUE but this is not the correct. Might be mistaken for overriding(CONFIG_NAME, CONFIG_VALUE).

**Review by kselveliev — APPROVED** [review]

File: hedera-node/hapi-fees/src/main/java/com/hedera/node/app/hapi/fees/usage/crypto/CryptoOpsUsage.java [hunk 2]
```diff
@@ -192,13 +192,13 @@ public void cryptoCreateUsage(
...
         if (maxAutomaticTokenAssociations > 0) {
             baseSize += INT_SIZE;
+            accumulator.addRbs(maxAutomaticTokenAssociations * lifeTime * CREATE_SLOT_MULTIPLIER);
         }
 
         /* Variable bytes plus two additional longs for balance and auto-renew period;
         plus a boolean for receiver sig required. */
         accumulator.addBpt(baseSize + 2 * LONG_SIZE + BOOL_SIZE);
         accumulator.addRbs((CRYPTO_ENTITY_SIZES.fixedBytesInAccountRepr() + baseSize) * lifeTime);
-        accumulator.addRbs(maxAutomaticTokenAssociations * lifeTime * CREATE_SLOT_MULTIPLIER);
```
Comment (MrValioBg) [inline hedera-node/hapi-fees/src/main/java/com/hedera/node/app/hapi/fees/usage/crypto/CryptoOpsUsage.java:201]: is this change needed? If the maxAutomaticTokenAssociations is 0, it would just call accumulator.addRbs(0) anyways

File: hedera-node/hapi-fees/src/main/java/com/hedera/node/app/hapi/fees/usage/crypto/CryptoOpsUsage.java [hunk 3]
```diff
@@ -192,13 +192,13 @@ public void cryptoCreateUsage(
 
         if (maxAutomaticTokenAssociations > 0) {
             baseSize += INT_SIZE;
+            accumulator.addRbs(maxAutomaticTokenAssociations * lifeTime * CREATE_SLOT_MULTIPLIER);
```
Comment (netopyr) [inline hedera-node/hapi-fees/src/main/java/com/hedera/node/app/hapi/fees/usage/crypto/CryptoOpsUsage.java:195]: In my understanding, **all** automatic token association slots should be free.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoCreateValidator.java [hunk 4]
```diff
@@ -100,7 +100,7 @@ public boolean tooManyAutoAssociations(
             @NonNull final TokensConfig tokensConfig) {
         return (entitiesConfig.limitTokenAssociations() && n > tokensConfig.maxPerAccount())
                 || n > ledgerConfig.maxAutoAssociations()
-                || (n < 0 && !entitiesConfig.unlimitedAutoAssociationsEnabled())
```
Comment (netopyr) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoCreateValidator.java:103]: Please update the Javadoc of this method. We are not only checking the upper bound.

File: hedera-node/hapi-fees/src/main/java/com/hedera/node/app/hapi/fees/usage/crypto/CryptoOpsUsage.java (hunk 3 above)
Comment (stoyanov-st) [reply hedera-node/hapi-fees/src/main/java/com/hedera/node/app/hapi/fees/usage/crypto/CryptoOpsUsage.java:195]: Do we want to completely remove this as I don't see a way to check the feature flag there?

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoCreateValidator.java (hunk 4 above)
Comment (stoyanov-st) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoCreateValidator.java:103]: Javadoc updated to reflect the changes

**Comment by stoyanov-st** [comment]
Merging this PR, any further comments can be addressed in the  [base PR](https://github.com/hashgraph/hedera-services/pull/13687)
